tests/NSubstitute.Acceptance.Specs/FormattingCallsWhenThrowingReceivedCallsExceptions.cs

```diff
@@ -312,40 +312,40 @@ protected override void ExpectedCall()
             [Test]
             public void Should_show_expected_call()
             {
-                ExceptionMessageContains("IntParamsMethod(1, 200, 300)");
+                ExceptionMessageContains("IntParamsMethod([1, 200, 300])");
             }
 
             [Test]
             public void Should_show_non_matching_calls_with_params_expanded()
             {
-                ExceptionMessageContains("IntParamsMethod(1, *2*, *3*)");
-                ExceptionMessageContains("IntParamsMethod(*4*, *5*)");
-                ExceptionMessageContains("IntParamsMethod(**)");
+                ExceptionMessageContains("IntParamsMethod([1, *2*, *3*])");
+                ExceptionMessageContains("IntParamsMethod([*4*, *5*])");
+                ExceptionMessageContains("IntParamsMethod([**])");
             }
         }
 
         public class When_checking_call_to_method_with_params_specified_as_an_array : Context
         {
             protected override void ConfigureContext()
             {
-                Sample.ParamsMethod(2, "hello", "everybody");
+                Sample.ParamsMethod(2, new[] { "hello", "everybody" });
             }
 
             protected override void ExpectedCall()
             {
-                Sample.Received().ParamsMethod(1, Arg.Is(new[] {"hello", "world"}));
```

Comment:
This test is intended to check matching when params are specified as an array (using `Arg.Is(new [] { ... })`).

Reply:
> Hi @riezebosch, thanks very much for this PR!
> 
> I like the general idea of unwrapping arrays (the messages become much nicer), but I think I'd prefer to keep the `params` calls reported using the same syntax as used by the caller. If I write `sub.Received().ParamsMethod(1, 2, 3)` I think it is confusing to see the error reported as "Expected `ParamsMethod([1, 2, 3])`" rather than as what I initially wrote.
> 
> @zvirja or @alexandrnikitin, can we please get your view on this too? Firstly on the general unwrapping enumerables, and secondly on the formatting of the params args.
> 
> Thanks,
> David

I'm experimenting with passing the `isParams` into the `ArrayContentsArgumentMatcher` and format the arguments accordingly. But what will become difficult then is rendering the arguments when the parameter is `params` but the argument is passed as an array.

In the end, a params argument is just an array. Wouldn't you expect from (the I suspect advanced users of such framework) to just know that and interpret the message accordingly?

Reply:
Good catch, this was not intended and revealed that the output is not what I was expecting.